odoo/fields.py

```diff
@@ -99,163 +99,66 @@ def __init__(cls, name, bases, attrs):
+    :type default: value or callable
 
-        :param states: a dictionary mapping state values to lists of UI attribute-value
-            pairs; possible attributes are: 'readonly', 'required', 'invisible'.
-            Note: Any state-based condition requires the ``state`` field value to be
+    :param dict states: a dictionary mapping state values to lists of UI attribute-value
+        pairs; possible attributes are: ``readonly``, ``required``, ``invisible``.
+
+        .. warning:: Any state-based condition requires the ``state`` field value to be
             available on the client-side UI. This is typically done by including it in
             the relevant views, possibly made invisible if not relevant for the
             end-user.
 
-        :param groups: comma-separated list of group xml ids (string); this
-            restricts the field access to the users of the given groups only
-
-        :param bool copy: whether the field value should be copied when the record
-            is duplicated (default: ``True`` for normal fields, ``False`` for
-            ``one2many`` and computed fields, including property fields and
-            related fields)
-
-        .. _field-computed:
-
-        .. rubric:: Computed fields
-
-        One can define a field whose value is computed instead of simply being
-        read from the database. The attributes that are specific to computed
-        fields are given below. To define such a field, simply provide a value
-        for the attribute ``compute``.
-
-        :param compute: name of a method that computes the field
-
-        :param inverse: name of a method that inverses the field (optional)
-
-        :param search: name of a method that implement search on the field (optional)
-
-        :param store: whether the field is stored in database (boolean, by
-            default ``False`` on computed fields)
-
-        :param compute_sudo: whether the field should be recomputed as superuser
-            to bypass access rights (boolean, by default ``True``)
-
-        The methods given for ``compute``, ``inverse`` and ``search`` are model
-        methods. Their signature is shown in the following example::
-
-            upper = fields.Char(compute='_compute_upper',
-                                inverse='_inverse_upper',
-                                search='_search_upper')
-
-            @api.depends('name')
-            def _compute_upper(self):
-                for rec in self:
-                    rec.upper = rec.name.upper() if rec.name else False
+    :param str groups: comma-separated list of group xml ids (string); this
+        restricts the field access to the users of the given groups only
 
-            def _inverse_upper(self):
-                for rec in self:
-                    rec.name = rec.upper.lower() if rec.upper else False
+    :param bool company_dependent: whether the field is company-dependent
```

Comment:
perhaps more explanations should be included here, since the notion of a company-dependent field is not at all obvious

Reply:
```rst
Company_dependent fields are stored as ir.property and have a fixed value by company for all records::

a.with_context(force_company=company1) == b.with_context(force_company=company1)
a.with_context(force_company=company2) == b.with_context(force_company=company2)
```

Or am I not understanding it correctly ? :thinking: